### ISSUE TITLE
Allow to specify TAA method

### ISSUE THREAD

**Issue by TimoGlastra** [hyperledger/aries-framework-javascript#707]
Currently we always take the first TAA acceptance mechanism. We should update this to be configurable so users of the framework can indicate which TAA acceptance mechanism to use

Relevant code:

https://github.com/hyperledger/aries-framework-javascript/blob/858253a2ed9da8f62a7b41b7a4c9c16dc9d2614c/packages/core/src/modules/ledger/services/IndyLedgerService.ts#L454-L478

As we have multi ledger support I propose we extend the indy ledgers config and add an optional taaAccept property to it with a version (you have to know which version of the TAA you've accepted).

```diff
// Add new interface
export interface TransactionAuthorAgreement {
    version: `${number}.${number}`
    acceptanceMechanism: string
}

export interface IndyPoolConfig {
  genesisPath?: string
  genesisTransactions?: string
  id: string
  isProduction: boolean
  // add optional property
  transactionAuthorAgreement?: TransactionAuthorAgreement
}
```

If the TAA hasn't been specified we should throw an error instead of automatically taking the first one. We need to keep it optional for agents that do not write to the ledger.

**Comment by morrieinmaas** [comment]
@TimoGlastra can you assign me this issue please

**Comment by TimoGlastra** [comment]
Fixed in #849
